lib/internal/process/esm_loader.js

```diff
@@ -3,15 +3,22 @@
 const {
   callbackMap,
 } = internalBinding('module_wrap');
+const {
+  ERR_INVALID_TYPE_FLAG,
+  ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING,
+} = require('internal/errors').codes;
+const { emitExperimentalWarning } = require('internal/util');
+
+const type = require('internal/options').getOptionValue('--entry-type');
+if (type && type !== 'commonjs' && type !== 'module')
+  throw new ERR_INVALID_TYPE_FLAG(type);
```

Comment:
We should push errors in the option parser instead of waiting until here. The JS land should be able to get this through `getOptionValue('--entry-type')` directly, this is not really expensive so there isn't a lot of value caching it.